src/main/java/com/github/mproberts/rxtools/map/SubjectMap.java

```diff
@@ -331,7 +331,64 @@ public void runIfBound(K key, final Runnable runnable)
         }
 
         if (isBound) {
-            runnable.run();
+            processNewFaultForKey(key);
+        }
+    }
+
+    private void processNewFaultForKey(final K key)
+    {
+        Completable.defer(new Callable<CompletableSource>() {
```

Comment:
why use a completable if you're going to subscribe to it right away on the default scheduler?